src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -4,86 +4,65 @@
+	
+	private MotorControllerGroup leftMotors;
+	private MotorControllerGroup rightMotors;
+	private DifferentialDrive drive;
 
 	public Drivetrain(CommandXboxController controller) {
 		this.controller = controller;
 
-		frontLeft = new CANSparkMax(0, MotorType.kBrushless);
-		frontRight = new CANSparkMax(1, MotorType.kBrushless);
-		backLeft = new CANSparkMax(2, MotorType.kBrushless);
-		backRight = new CANSparkMax(3, MotorType.kBrushless);
+		frontLeft = new Spark(0);
+		frontRight = new Spark(2);
+		backLeft = new Spark(1);
+		backRight = new Spark(3);
+
+		leftMotors = new MotorControllerGroup(frontLeft,backLeft);
+		rightMotors = new MotorControllerGroup(frontRight, backRight);
+		
+		rightMotors.setInverted(true);
+		
+		drive = new DifferentialDrive(leftMotors, rightMotors);
 
-		frontRight.setInverted(true);
-		backRight.setInverted(true);
 	}
 
 	@Override
 	public void periodic() {
-		double joyX = controller.getRawAxis(OperatorConstants.XBOX_LEFT_X_AXIS);
-		double joyY = -controller.getRawAxis(OperatorConstants.XBOX_LEFT_Y_AXIS);
-		double rotation = ControlConstants.ROTATION_MULT
-				* (controller.getRightTriggerAxis() - controller.getLeftTriggerAxis());
+		double move = controller.getLeftTriggerAxis();
+		double turn = controller.getLeftX();
 
-		mecanumDrive(joyX, joyY, rotation);
+		drive.arcadeDrive(move, turn);
 	}
 
-	// 2020 mecanum drive code
-	public void mecanumDrive(double joystickX, double joystickY, double rotation) {
-		// Deadband inputs
-		if (Math.abs(rotation) < ControlConstants.ROTATION_DEADBAND)
-			rotation = 0;
-		if (Math.abs(joystickX) < ControlConstants.JOY_DEADBAND)
-			joystickX = 0;
-		if (Math.abs(joystickY) < ControlConstants.JOY_DEADBAND)
-			joystickY = 0;
-
-		// Cap rotation to ControlConstants value
-		if (Math.abs(rotation) > ControlConstants.MAX_TURN_SPEED)
-			rotation *= ControlConstants.MAX_TURN_SPEED / Math.abs(rotation);
-
-		// Calculate speed for each wheel
-		double frontRightPower = joystickY - joystickX - rotation;
-		double frontLeftPower = joystickY + joystickX + rotation;
-		double backLeftPower = joystickY - joystickX + rotation;
-		double backRightPower = joystickY + joystickX - rotation;
-
-		// Cap motor powers
-		if (Math.abs(frontLeftPower) > ControlConstants.MAX_ROBOT_SPEED)
-			frontLeftPower *= ControlConstants.MAX_ROBOT_SPEED / Math.abs(frontLeftPower);
-		if (Math.abs(frontRightPower) > ControlConstants.MAX_ROBOT_SPEED)
-			frontRightPower *= ControlConstants.MAX_ROBOT_SPEED / Math.abs(frontRightPower);
-		if (Math.abs(backLeftPower) > ControlConstants.MAX_ROBOT_SPEED)
-			backLeftPower *= ControlConstants.MAX_ROBOT_SPEED / Math.abs(backLeftPower);
-		if (Math.abs(backRightPower) > ControlConstants.MAX_ROBOT_SPEED)
-			backRightPower *= ControlConstants.MAX_ROBOT_SPEED / Math.abs(backRightPower);
-
-		// Power the motors
-		frontLeft.set(frontLeftPower);
-		frontRight.set(frontRightPower);
-		backLeft.set(backLeftPower);
-		backRight.set(backRightPower);
-	}
+
```

Comment:
we should add a method here that allows a separate command (eg LimelightFollow) can also access the Drivetrain.
Similar to the mecanumDrive code deleted above